### PR TITLE
feat: Added Four Channel Mode in Logic Analyzer

### LINKED ISSUE #1167: Four Channel Mode not working in Logic Analyzer
**Actual Behaviour**

`Four Channel Mode` isn't working in Logic Analyzer

**Expected Behaviour**

It should work

**Steps to reproduce it**

Go to Logic Analyzer -> Switch to `Four Channel Mode` -> Analyze

**LogCat for the issue**

Not applicable

**Screenshots of the issue**

Not applicable

**Would you like to work on the issue?**

Yes

### PR DESCRIPTION
**Pull request by harsh-2711**
Fixes #1167 

**Changes**: Added Four Channel Mode in Logic Analyzer instrument

**Checklist**: [Please tick following check boxes with `[x]` if the respective task is completed]
- [x] I have used resources from `strings.xml`, `dimens.xml` and `colors.xml` without hard-coding them
- [x] No modifications done at the end of resource files `strings.xml`, `dimens.xml` or `colors.xml`
- [x] I have reformatted code in every file included in this PR [<kbd>CTRL</kbd>+<kbd>ALT</kbd>+<kbd>L</kbd>]
- [x] My code does not contain any extra lines or extra spaces
- [x] I have requested reviews from other members

**APK for testing**: 
[app-debug.zip](https://github.com/fossasia/pslab-android/files/2177994/app-debug.zip)

NOTE: Please merge PR #1159 before merging this PR

### CODE REVIEW

**Comment by harsh-2711** [comment]
@abhinavraj23 I can't squash the commits. It will reduce to one once PR #1159 is merged.

**Comment by CloudyPadmal** [comment]
Yes @abhinavraj23 , for now we can keep this PR as it is and when #1159 is merged, @harsh-2711 can update this right?

**Comment by harsh-2711** [comment]
@CloudyPadmal Please remove `WIP` tag. @Vikum94 I have requested @abhinavraj23 for screenshots of both Three Channel Mode and Four Channel Mode as my device isn't working properly. Will update by tomorrow.

**Comment by harsh-2711** [comment]
@CloudyPadmal Please remove conflicts tag

File: app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java [hunk 1]
```diff
@@ -313,109 +315,169 @@ public void onPageScrollStateChanged(int state) {
...
 
-                        @Override
-                        public void onNothingSelected() {
+                        // Setting cursor to display time at highlighted points
+                        listener = new OnChartValueSelectedListener() {
+                            @Override
+                            public void onValueSelected(Entry e, Highlight h) {
+                                double result = Math.round(e.getX() * 100.0) / 100.0;
+                                xCoordinateText.setText("Time:  " + String.valueOf(result) + " mS");
+                                Log.i("Entry selected", e.toString());
```
Comment (CloudyPadmal) [inline app/src/main/java/org/fossasia/pslab/fragment/LALogicLinesFragment.java]: Do you need this logging to test? If not we can remove it.

**Comment by abhinavraj23** [comment]
@CloudyPadmal Have you tested the apk , is the four channel mode working perfectly in your device?

**Comment by abhinavraj23** [comment]
Also @harsh-2711 please add proper title i.e feat, refactor, fix etc..

**Comment by harsh-2711** [comment]
@CloudyPadmal Changes done. Please review.

**Comment by harsh-2711** [comment]
@CloudyPadmal @abhinavraj23 All changes done. Please review it.

**Pull request merged**